### PR TITLE
Improved getting typed registered variables and path type changed

### PR DESCRIPTION
**Pull request by anki-code**
Hi! xonsh is great!

In this PR:

1. The `path` type in `${...}.register` was renamed to `env_path` as it should be and added new `path` type instead that represent `pathlib.Path`. Now you can register typed environment variables that will be converted to `Path` objects.

2. Fixed getting typed registered environment variable when it was initialized before registration:

    ```python
    $QWE='/tmp'
    ${...}.register(name='QWE', type='path')
    $QWE
    # Before fix: return string '/tmp'
    # After fix: return PosixPath('/tmp') and immediately do $QWE.exists()
    ```

3. Tests added.

### CODE REVIEW

**Comment by anki-code** [comment]
@scopatz I'm done, please merge ;)

File: xonsh/tools.py [hunk 1]
```diff
@@ -1171,11 +1171,22 @@ def ensure_string(x):
...
+    """This tests if something is a path."""
+    return isinstance(x, pathlib.Path)
+
+
 def is_env_path(x):
     """This tests if something is an environment path, ie a list of strings."""
     return isinstance(x, EnvPath)
 
 
+def str_to_path(x):
```
Comment (scopatz) [inline xonsh/tools.py:1184]: Can you please add some tests to test_tools.py for the functions you added here.

**Comment by scopatz** [comment]
Thanks! I think this just needs a few more tests

File: xonsh/tools.py (hunk 1 above)
Comment (anki-code) [reply xonsh/tools.py:1184]: Done )

**Comment by anki-code** [comment]
@scopatz a few more tests done )

**Comment by scopatz** [comment]
Thanks a ton @anki-code!

**Pull request merged**